docs/sdk.rst

```diff
@@ -0,0 +1,191 @@
+access control on both functions and endpoints.
+Globus Auth is an identity and access management platform that provides authentication brokering
+capablities enabling users to login using one of several hundred supported identities.
+It also provides group and profile management for user accounts.
+As part of the authentication process, funcX will request access
+to your identity (to retrieve your email address) and Globus Groups. funcX uses
+Groups to facilitate sharing and to make authorization decisions.
+funcX allows endpoints and functions to be shared by associating a Globus Group.
+
+.. note:: funcX internally caches function, endpoint, and authorization lookups. Caches are based on user authentication tokens. To force refresh cached
+          entries you can re-authenticate your client with `force_login=True`.
+
+Registering Functions
+---------------------
+
+You can register a Python function with funcX via `register_function()`. Function registration serializes the
+function body and transmits it to funcX. Once the function is registered with funcX, it is assigned a
+UUID that can be used to manage and invoke the function.
+
+.. note:: You must import any dependencies required by the function inside the function body.
+
+
+The following example shows how to register a function. In this case the function simply
+returns the platform information of the system on which it is executed. The function
+is defined in the same way as any Python function before being registered with funcX.
+
+.. code-block:: python
+
+  def platform_func():
+    import platform
+    return platform.platform()
+
+  func_uuid = fxc.register_function(platform_func)
+
+
+Running Functions
+-----------------
+You can invoke a function using the UUID returned when registering the function. The `run()` function
+requires that you specify the function (`function_id`) and endpoint (`endpoint_id`) on which to execute
+the function. funcX will return a UUID for the executing function (called a task) via which you can
+monitor status and retrieve results.
+
+.. code-block:: python
+
+  tutorial_endpoint = '4b116d3c-1703-4f8f-9f6f-39921e5864df'
+  task_id = fxc.run(endpoint_id=tutorial_endpoint, function_id=func_uuid)
+
+
+Retrieving Results
+-------------------
+The result of your function's invocation can be retrieved using the `get_result()` function. This will either
+return the deserialized result of your invocation or raise an exception indicating that the
+task is still pending.
+
+.. note:: If your function raises an exception, get_result() will reraise it.
+
+.. code-block:: python
+
+  try:
+    print(fxc.get_result(task_id))
+  except Exception as e:
+    print("Exception: {}".format(e))
+
+.. note:: funcX caches results in the cloud until they have been retrieved. The SDK also caches results
+          during a session. However, calling `get_result()` from a new session will not be able to access the results.
```

Comment:
I'm confused by this - why not?

Reply:
Once a result is fetched from our hosted services we garbage collect to save memory.